python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -737,15 +743,35 @@ def _resolve_and_update_nested_resources(
         to_update = {**resources_to_update, **partial_resources_to_update}
         return self._with_updated_values(to_update)
 
+    def with_resource_context(
+        self, resource_context: InitResourceContext
+    ) -> "ConfigurableResourceFactory[TResValue]":
+        """Returns a new instance of the resource with the given resource init context bound."""
+        copy = self._with_updated_values({})
+        copy._state__internal__ = copy._state__internal__._replace(  # noqa: SLF001
+            resource_context=resource_context
+        )
+        return copy
+
     def initialize_and_run(self, context: InitResourceContext) -> TResValue:
```

Comment:
not your diff, but should we make this private since we don't want users to call this directly?